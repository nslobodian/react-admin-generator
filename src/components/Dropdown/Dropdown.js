import React from 'react'
import { withState } from 'recompose'
import cn from 'classnames'
import s from './Dropdown.scss'

export const Dropdown = withState('dropdown', 'toggleDropdown', false)(({ dropdown, toggleDropdown, children, title, type, noHideOnSelect }) =>
  <div className={cn('btn-group btn-block', s.Dropdown)}>
    <div className={cn('dropdown', dropdown && 'open')}>
      <button className={`btn btn-${type} dropdown-toggle btn-block`}
        type='button' data-toggle='dropdown' aria-haspopup='true'
        aria-expanded={dropdown} onClick={() => toggleDropdown(v => !v)}>
        <div className='pull-left'>{title}</div>
        <div className={cn('pull-right caret', s.caret)} />
      </button>
      <ul className='dropdown-menu btn-block' onClick={() => !noHideOnSelect && toggleDropdown(v => !v)}>
        {children}
      </ul>
    </div>
  </div>
)

export default Dropdown
