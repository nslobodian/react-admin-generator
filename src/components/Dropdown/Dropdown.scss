@import '~styles/shared';

.Dropdown {
  button {
    height: $input-height-base;

    .caret {
      margin-top: 5px
    }
  }
}

.multiValue {
  :global {
    .Select-value-label {
      font-size: 9px;
    }
  }
}
