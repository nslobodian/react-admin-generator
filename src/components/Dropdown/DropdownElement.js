import React from 'react'
import Select from 'react-select'
import pure from 'recompose/pure'
import s from './Dropdown.scss'

export default pure(({ input: { value, onChange, onBlur, name }, meta: { touched, invalid, error }, label, forceShowErrors, blurable = true, style, ...props }) =>
  <div style={style}>
    {label && <span>{label}</span>}
    <Select value={value} searchable={false} clearable={false} {...props}
      onChange={val => {
        onChange(val)
        blurable && onBlur(val)
      }}
      className={props.multi && s.multiValue} />
    {(touched || forceShowErrors) && invalid &&
    <div className='alert alert-danger m-t-1' role='alert'>{error}</div>}
  </div>
)
