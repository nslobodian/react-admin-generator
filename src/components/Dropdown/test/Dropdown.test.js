import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Dropdown from '../Dropdown'

describe('(Components) Dropdown', () => {
  it('Should render Dropdown properly', () => {
    const props = {
      children: [],
      dropdown: true,
      toggleDropdown: jest.fn(),
      title: 'title',
      type: 'TYPE',
      noHideOnSelect: true,
    }
    const wrapper = shallow(
      <Dropdown
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find('ul').simulate('click')
    wrapper.find('button').simulate('click')
  }),
  it('Should render Dropdown properly without some attributes', () => {
    const props = {
      children: [],
      toggleDropdown: jest.fn(),
      title: 'title',
      type: 'TYPE',
      noHideOnSelect: false,
    }
    const wrapper = shallow(
      <Dropdown
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find('ul').simulate('click')
    wrapper.find('button').simulate('click')
  })
})
