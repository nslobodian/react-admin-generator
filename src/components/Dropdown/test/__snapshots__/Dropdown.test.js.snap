// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) Dropdown Should render Dropdown properly 1`] = `
<div
  className="btn-group btn-block Dropdown"
>
  <div
    className="dropdown"
  >
    <button
      aria-expanded={false}
      aria-haspopup="true"
      className="btn btn-TYPE dropdown-toggle btn-block"
      data-toggle="dropdown"
      onClick={[Function]}
      type="button"
    >
      <div
        className="pull-left"
      >
        title
      </div>
      <div
        className="pull-right caret caret"
      />
    </button>
    <ul
      className="dropdown-menu btn-block"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`(Components) Dropdown Should render Dropdown properly without some attributes 1`] = `
<div
  className="btn-group btn-block Dropdown"
>
  <div
    className="dropdown"
  >
    <button
      aria-expanded={false}
      aria-haspopup="true"
      className="btn btn-TYPE dropdown-toggle btn-block"
      data-toggle="dropdown"
      onClick={[Function]}
      type="button"
    >
      <div
        className="pull-left"
      >
        title
      </div>
      <div
        className="pull-right caret caret"
      />
    </button>
    <ul
      className="dropdown-menu btn-block"
      onClick={[Function]}
    />
  </div>
</div>
`;
